### PR TITLE
Trigger releases from GitHub

### PR DESCRIPTION
**Pull request by slewis74**
Refactoring the pipeline a bit. In the modified model

- PRs are built and published to Octopus
- creating a release in GitHub triggers a build and publish to a channel that can publish to production. Release notes included on the GitHub release will be carried through to Octopus' release

Future work may include using [release-please](https://github.com/googleapis/release-please) as the trigger to then create the release in GitHub.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -58,41 +74,70 @@ jobs:
...
-          tar -czf OctoTFS.publish.$OCTOPUS_VERSION.tar.gz ./publish.ps1 ./dist/extension-manifest*.json
-      - name: Push Package
-        run: |
-          docker run -e "OCTOPUS_CLI_SERVER=${{ secrets.OCTOPUS_URL }}" -e "OCTOPUS_CLI_API_KEY=${{ secrets.INTEGRATIONS_API_KEY }}" -v $(pwd):/src octopusdeploy/octo push --space "Integrations" --package OctoTFS.vsix.$OCTOPUS_VERSION.tar.gz --package OctoTFS.publish.$OCTOPUS_VERSION.tar.gz --overwrite-mode OverwriteExisting
-      - name: Create Release
+          tar -czf OctoTFS.vsix.${{ needs.build.outputs.package_version }}.tar.gz ./dist/Artifacts/**/*.vsix
+          tar -czf OctoTFS.publish.${{ needs.build.outputs.package_version }}.tar.gz ./publish.ps1 ./dist/extension-manifest*.json
+
+      - name: Install Octopus CLI 🐙
+        uses: OctopusDeploy/install-octopus-cli-action@v1.1.7
```
Comment (jbristowe) [inline .github/workflows/build.yml:117]: `install-octopus-cli-action` v1.2 is now available

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -58,41 +74,70 @@ jobs:
...
         with:
           name: dist
           path: dist
+
       - name: Use Node.js
         uses: actions/setup-node@v2
         with:
           node-version: '16'
           cache: 'npm'
+
```
Comment (jbristowe) [inline .github/workflows/build.yml:94]: Why install Go?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -58,41 +74,70 @@ jobs:
...
-    needs: test
+    needs: [build, test]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/download-artifact@v2
         with:
           name: dist
           path: dist
+
```
Comment (jbristowe) [inline .github/workflows/build.yml:88]: `actions/checkout` v3 is available

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -58,41 +74,70 @@ jobs:
...
+          tar -czf OctoTFS.publish.${{ needs.build.outputs.package_version }}.tar.gz ./publish.ps1 ./dist/extension-manifest*.json
+
+      - name: Install Octopus CLI 🐙
+        uses: OctopusDeploy/install-octopus-cli-action@v1.1.7
+
+      - name: Push Package 🐙
+        uses: OctopusDeploy/push-package-action@v1.1.2
+        with:
+            space: "Integrations"
+            packages: "OctoTFS.vsix.${{ needs.build.outputs.package_version }}.tar.gz,OctoTFS.publish.${{ needs.build.outputs.package_version }}.tar.gz"
```
Comment (jbristowe) [inline .github/workflows/build.yml:123]: Better to use multiple lines for readability:

```yaml
packages: |
  OctoTFS.vsix.${{ needs.build.outputs.package_version }}.tar.gz
  OctoTFS.publish.${{ needs.build.outputs.package_version }}.tar.gz
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (slewis74) [reply .github/workflows/build.yml]: Updated 👍

File: .github/workflows/build.yml (hunk 2 above)
Comment (slewis74) [reply .github/workflows/build.yml:94]: I think this was a left over from early versions running node-prune to clean up. This looks to be taken care of ootb now, so I've removed it from here and the pack.ps1 file no longer calls go to run `node-prune`

**Review by jbristowe — APPROVED** [review]

**Pull request merged**